### PR TITLE
Updated controllers to run with the latest version of SAM in Stonefish

### PR DESCRIPTION
**Pull request by svbhat**

### CODE REVIEW

**Comment by svbhat** [comment]
Updates to controller coefficients, and added parts to CMakelists.txt to allow installation.

**Comment by nilsbore** [comment]
@svbhat Have you tried running this from an install space?

File: sam_basic_controllers/src/pid_utils/odom_listener.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ void OdomCallback(const nav_msgs::Odometry& odom_msg)
     current_pitch.data= p;
```
Comment (nilsbore) [inline sam_basic_controllers/src/pid_utils/odom_listener.cpp:30]: Wasn't this replaced by the [tf_convenience_topics](https://github.com/smarc-project/smarc_navigation/tree/noetic-devel/tf_convenience_topics/src) package??

File: sam_basic_controllers/launch/SIM/static_controllers_SIM.launch [hunk 2]
```diff
@@ -10,7 +10,7 @@
     <group ns="ctrl">
 
         <node name="odom_listener" pkg="sam_basic_controllers" type="odom_listener" output="screen" >
```
Comment (nilsbore) [inline sam_basic_controllers/launch/SIM/static_controllers_SIM.launch]: Same question here basically, shouldn't we use [tf_convenience_topics](https://github.com/smarc-project/smarc_navigation/tree/noetic-devel/tf_convenience_topics/src) instead?

File: sam_basic_controllers/src/pid_utils/odom_listener.cpp (hunk 1 above)
Comment (svbhat) [reply sam_basic_controllers/src/pid_utils/odom_listener.cpp:30]: I have updated it now to use the tf_convenience_topics package. I haven't yet removed the odom_listener from the controllers, but I will do so once we test things with the tf_convenience_topics package.

**Comment by svbhat** [comment]
I'm getting an error with CI saying it cannot locate the dependency for sbg_driver in sam_dead_reckoning, though it is in the package.xml and in the rosinstall file for smarc_navigation.

**Comment by svbhat** [comment]
It now passes CI after I included dependencies for smarc_navigation as well in the rosinstall file.

File: sam_basic_controllers/launch/SIM/static_controllers_SIM.launch [hunk 3]
```diff
@@ -23,12 +23,18 @@
...
-            <param name="loop_freq" value="50" />
-        </node-->
+        <!--node name="tf_listener" pkg="tf_convenience_topics" type="tf_listener" output="screen" >
+	    	<param name="world_frame" value="/utm"/>	    	
+	    	<param name="base_frame" value="/$(arg robot_name)/base_link"/>
+	        <param name="topic_from_dvl" value="$(arg robot_name)/core/dvl"/>
+            	<param name="loop_freq" value="50" />
+    	</node-->
+
+        <node name="odom_listener" pkg="tf_convenience_topics" type="odom_listener" output="screen" >
```
Comment (nilsbore) [inline sam_basic_controllers/launch/SIM/static_controllers_SIM.launch]: I think this is OK for now, but actually I think we should have the `tf_convenience_topics` be started as part of the DR or part of the sim.

Comment (svbhat) [reply sam_basic_controllers/launch/SIM/static_controllers_SIM.launch]: Ok, now the controllers listen to sam/dr/odom_listener/x_feedback etc.., and the odom_listener node is launched from tf_convenience_topics. That launch can be included with DR. I have made the update now and it passes CI.

Comment (nilsbore) [reply sam_basic_controllers/launch/SIM/static_controllers_SIM.launch]: Ok, I guess the naming doesn't really matter here but some parts should be the same as in https://github.com/smarc-project/smarc_msgs/pull/22 . Is that the case?

**Pull request merged**